hazelcast/src/main/java/com/hazelcast/sql/impl/exec/scan/index/IndexInFilter.java

```diff
@@ -70,8 +72,21 @@ public IndexInFilter(List<IndexFilter> filters) {
     }
 
     @Override
-    public Iterator<QueryableEntry> getEntries(InternalIndex index, ExpressionEvalContext evalContext) {
-        Map<Comparable, IndexFilter> canonicalFilters = new HashMap<>();
+    public Iterator<QueryableEntry> getEntries(InternalIndex index, boolean descending, ExpressionEvalContext evalContext) {
+
+        // Sort the filter Comparables
+        // NULLs are coming last if the sort is ASC
+        // NULLs are coming first if the sort is DESC
+        NavigableMap<Comparable, IndexFilter> canonicalFilters = new TreeMap<>((o1, o2) -> {
+            if (o1 == NULL) {
```

Comment:
In `IndexInFilter`, we put `NULL`s last for `ASC`. But in SortKeyComparator we have the following code, which AFAIU puts `NULL`s first for `ASC`:
```
if (o1 == null) {
    return asc ? -1 : 1;
}
```

What is the canonical implementation that we would like to have? Do we have tests that cover all types of filters, as well as distributed vs non-distirubted case? Another question - does composite filters handle nulls in the same way as we expect?

Reply:
That is a leftover because we changed our decision and now support the same NULLs collation as in the composite index. I fixed it in the https://github.com/hazelcast/hazelcast/pull/18013/commits/5a7d73a49f54ca8fb2bb26149d7dfee9242730f7 as well as individual unit tests.